storm-core/src/clj/org/apache/storm/daemon/nimbus.clj

```diff
@@ -1355,48 +1366,843 @@
 (defmethod blob-sync :local [conf nimbus]
   nil)
 
-(defserverfn service-handler [conf inimbus]
-  (.prepare inimbus conf (master-inimbus-dir conf))
-  (log-message "Starting Nimbus with conf " conf)
-  (let [nimbus (nimbus-data conf inimbus)
-        blob-store (:blob-store nimbus)
-        principal-to-local (AuthUtils/GetPrincipalToLocalPlugin conf)
+(defn extract-cluster-metrics [^ClusterSummary summ]
```

Comment:
since we are doing this in ui isn't it better move it to helpers or utils and import that here in nimbus